### PR TITLE
Add `main` flag to threads and `in_foreground` flag for app contexts

### PR DESCRIPTION
**Pull request by marandaneto**
## :scroll: Description
Add main flag to threads and in_foreground flag for app contexts

## :bulb: Motivation and Context
Relates to https://github.com/getsentry/team-mobile/issues/36

## :pencil: Checklist

- [X] I reviewed the submitted code.
- [X] I added tests to verify the changes.
- [X] No new PII added or SDK only sends newly added PII if `sendDefaultPII` is enabled.
- [ ] I updated the docs if needed.
- [ ] Review from the native team if needed.
- [X] No breaking change or entry added to the changelog.
- [X] No breaking change for hybrid SDKs or communicated to hybrid SDKs.

### CODE REVIEW

**Comment by marandaneto** [comment]
@markushi @romtsn can I get early feedback here before going ahead?

File: sentry-android-core/src/main/java/io/sentry/android/core/DefaultAndroidEventProcessor.java [hunk 1]
```diff
@@ -256,21 +277,21 @@ private void setAppExtras(final @NotNull App app) {
     return Build.CPU_ABI2;
   }
 
-  @SuppressWarnings({"ObsoleteSdkInt", "deprecation"})
+  @SuppressWarnings({"ObsoleteSdkInt", "deprecation", "NewApi"})
   private void setArchitectures(final @NotNull Device device) {
-    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
-      String[] supportedAbis = Build.SUPPORTED_ABIS;
-      device.setArchs(supportedAbis);
+    String[] supportedAbis;
```
Comment (markushi) [inline sentry-android-core/src/main/java/io/sentry/android/core/DefaultAndroidEventProcessor.java:282]: Could be marked as `final`

**Comment by markushi** [comment]
Looking good 👍 , if you need any help on the TODOs around the thread `main` or `current` flags please let me know.

**Comment by romtsn** [comment]
lgtm as well

File: sentry-android-core/src/main/java/io/sentry/android/core/DefaultAndroidEventProcessor.java [hunk 2]
```diff
@@ -241,9 +252,19 @@ private void setDist(final @NotNull SentryBaseEvent event, final @NotNull String
     }
   }
 
-  private void setAppExtras(final @NotNull App app) {
+  private void setAppExtras(final @NotNull App app, final @NotNull Hint hint) {
     app.setAppName(getApplicationName());
     app.setAppStartTime(DateUtils.toUtilDate(AppStartState.getInstance().getAppStartTime()));
+
+    // This should not be set by Hybrid SDKs since they have their own app's lifecycle
+    if (!HintUtils.isFromHybridSdk(hint) && app.getInForeground() == null) {
```
Comment (marandaneto) [inline sentry-android-core/src/main/java/io/sentry/android/core/DefaultAndroidEventProcessor.java:260]: Running into https://github.com/getsentry/sentry-java/issues/2525

File: sentry-android-core/src/main/java/io/sentry/android/core/DefaultAndroidEventProcessor.java [hunk 3]
```diff
@@ -201,23 +201,34 @@ private void mergeOS(final @NotNull SentryBaseEvent event) {
...
+        final boolean isMainThread = AndroidMainThreadChecker.getInstance().isMainThread(thread);
+
+        // TODO: Fix https://github.com/getsentry/team-mobile/issues/47
         if (thread.isCurrent() == null) {
-          thread.setCurrent(AndroidMainThreadChecker.getInstance().isMainThread(thread));
+          thread.setCurrent(isMainThread);
+        }
+
+        // This should not be set by Hybrid SDKs since they have their own threading model
+        if (!isHybridSDK && thread.isMain() == null) {
```
Comment (marandaneto) [inline sentry-android-core/src/main/java/io/sentry/android/core/DefaultAndroidEventProcessor.java:230]: Similar to https://github.com/getsentry/sentry-java/pull/2516/files#r1099000230

**Comment by marandaneto** [comment]
@romtsn @markushi this is ready for final review.
I found that problem https://github.com/getsentry/sentry-java/issues/2525 but it's not a blocker since this check already exists in other places.
This likely has to be addressed as well but can be a separate PR.

**Review by markushi — APPROVED** [review]
Nice!

**Review by romtsn — APPROVED** [review]

**Pull request merged**